packages/expo-updates/e2e/__tests__/setup/project.js

```diff
@@ -0,0 +1,143 @@
+const spawnAsync = require('@expo/spawn-async');
+const fs = require('fs/promises');
+const path = require('path');
+
+async function setupAsync(workingDir, repoRoot, runtimeVersion) {
+  // initialize project
+  await spawnAsync('expo-cli', ['init', 'updates-e2e', '--yes'], {
+    cwd: workingDir,
+    stdio: 'inherit',
+  });
+  const projectRoot = path.join(workingDir, 'updates-e2e');
+
+  // add local dependencies
+  let packageJson = JSON.parse(await fs.readFile(path.join(projectRoot, 'package.json'), 'utf-8'));
+  packageJson = {
+    ...packageJson,
+    resolutions: {
+      ...(packageJson.resolutions ?? {}),
+      'expo-application': 'file:../expo/packages/expo-application',
+      'expo-constants': 'file:../expo/packages/expo-constants',
+      'expo-eas-client': 'file:../expo/packages/expo-eas-client',
+      'expo-error-recovery': 'file:../expo/packages/expo-error-recovery',
+      'expo-file-system': 'file:../expo/packages/expo-file-system',
+      'expo-font': 'file:../expo/packages/expo-font',
+      'expo-json-utils': 'file:../expo/packages/expo-json-utils',
+      'expo-keep-awake': 'file:../expo/packages/expo-keep-awake',
+      'expo-manifests': 'file:../expo/packages/expo-manifests',
+      'expo-modules-core': 'file:../expo/packages/expo-modules-core',
+      'expo-structured-headers': 'file:../expo/packages/expo-structured-headers',
+      'expo-updates-interface': 'file:../expo/packages/expo-updates-interface',
+    },
+  };
+  await fs.writeFile(
+    path.join(projectRoot, 'package.json'),
+    JSON.stringify(packageJson, null, 2),
+    'utf-8'
+  );
+  await spawnAsync(
+    'yarn',
+    [
+      'add',
+      'file:../expo/packages/expo-updates',
+      'file:../expo/packages/expo',
+      'file:../expo/packages/expo-splash-screen',
+      'file:../expo/packages/expo-status-bar',
+    ],
+    {
+      cwd: projectRoot,
+      stdio: 'inherit',
+    }
+  );
+
+  // configure app.json
+  let appJson = JSON.parse(await fs.readFile(path.join(projectRoot, 'app.json'), 'utf-8'));
+  appJson = {
+    ...appJson,
+    expo: {
+      ...(appJson.expo ?? {}),
+      name: 'updates-e2e',
+      runtimeVersion,
+      plugins: ['expo-updates'],
+      android: { ...(appJson.android ?? {}), package: 'dev.expo.updatese2e' },
+      ios: { ...(appJson.ios ?? {}), bundleIdentifier: 'dev.expo.updatese2e' },
+      updates: {
+        ...(appJson.updates ?? {}),
+        url: `http://${process.env.UPDATES_HOST}:${process.env.UPDATES_PORT}/update`,
+      },
+    },
+  };
+  await fs.writeFile(path.join(projectRoot, 'app.json'), JSON.stringify(appJson, null, 2), 'utf-8');
+
+  // generate and configure code signing
+  await spawnAsync(
+    'yarn',
+    [
+      'expo-updates',
+      'codesigning:generate',
+      '--key-output-directory',
+      'keys',
+      '--certificate-output-directory',
+      'certs',
+      '--certificate-validity-duration-years',
+      '1',
+      '--certificate-common-name',
+      '"E2E Test App"',
```

Comment:
Should be OK (and actually correct) to remove the quotes because Node will make sure each item is treated as a single argument value:
```suggestion
      'E2E Test App',
```